pulp_python/app/management/commands/repair-python-metadata.py

```diff
@@ -0,0 +1,107 @@
+import re
+import os
+from django.core.management import BaseCommand, CommandError
+from gettext import gettext as _
+
+from django.conf import settings
+
+from pulpcore.plugin.util import extract_pk
+from pulp_python.app.models import PythonPackageContent, PythonRepository
+from pulp_python.app.utils import artifact_to_python_content_data
+
+
+def repair_metadata(content):
+    """
+    Repairs the metadata for the passed in content queryset.
+    :param content: The PythonPackageContent queryset.
+    Return: number of content units that were repaired
+    """
+    # TODO: Add on_demand content repair?
```

Comment:
This data repair is not meant to unblock a migration / update, right?
In that case I'd say a best effort approach is just fine. We never really own the on-demand content anyway.

Reply:
No it's not meant to unblock, but I have a feeling that a lot of packages in users' repositories contain bad metadata. Like when we do a sync (which are default on-demand) currently all the non-latest versions of a package potentially get incorrect metadata because the majority of the available metadata (the `info` section) is from the latest package. There is a way to get the correct metadata for each version, but it adds an API call for each version, so I've never implemented it.

Reply:
How bad is it? Is it documented?

Reply:
It's not too hard, it's this api: https://docs.pypi.org/api/json/. You've seen it around, I've been using it in some of my scripts (oci-images check updates script for example)